llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp

```diff
@@ -2920,6 +2920,11 @@ bool MipsAsmParser::loadAndAddSymbolAddress(const MCExpr *SymExpr,
         (Res.getSymA()->getSymbol().isELF() &&
          cast<MCSymbolELF>(Res.getSymA()->getSymbol()).getBinding() ==
              ELF::STB_LOCAL);
+    if (!IsLocalSym && ABI.IsO32()) {
+      // PrivateGlobalPrefix for O32 is '$', while we support '.L' anyway.
```

Comment:
This comment is confusing.

// For O32, "$"-prefixed symbols are recognized as temporary while .L-prefixed symbols are not (PrivateGlobalPrefix is "$"). Recognize ".L" manually.